### PR TITLE
CLI tool to check redirects

### LINKED ISSUE #1307: Redirects needs to be pickier with weird names
I someone managed to upload  S3 (redirect) key with a blank name.
Screen Shot 2020-09-22 at 5 08 47 PM

I think the root cause is bad lines in the `_redirects.txt` file. 
Screen Shot 2020-09-22 at 5 10 23 PM

Perhaps the ideal thing is to fix the dumper so these aren't even in there in the first place.

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #1307
Part of #865
Part of #801

First and foremost, now, when it parses the `_redirects.txt` it does so in a MUCH stricter way. But it still has some work. 
For now, it checks that the lines are sane. Here are the rules so far:

* _redirects.txt file must have a trailing newline.
* Each line, split by whitespace should be exactly 2 parts
* The from-URL must start with `/$locale/docs/`
* That `$locale` must be one in `VALID_LOCALES` (e.g. `en-US`)

All of these rules seem to satisfy the `_redirects.txt` in the `main` branch, as of today, in mdn/content. 

Now, the CLI part of it is less important. For now. The intention is that we'll call this from the mdn/content repo in PR Builds CI. That way, if a PR author has messed with the `_redirects.txt` we'll run `yarn redirects validate` and if anything breaks the rules, it'll exit non-zero and the PR author should be able to see (although rather deep in the CI logs output) what was wrong and which line was the trouble. 

I say we land this (pending review) and then, start a new thread to incorporate some stricter rules that go deeper than just the strings on each line. For example, if I have this:

```
/en-US/docs/Foo /en-US/docs/Bar
/en-US/docs/Bar /en-US/docs/Buzz
```
...it should *not* fail validation but what should be stored in the `redirects` map is:

```
/en-US/docs/Foo   -->   /en-US/docs/Buzz
/en-US/docs/Bar   -->   /en-US/docs/Buzz
```

Secondly, if you have this:
```
/en-US/docs/Foo /en-US/docs/Bar
/en-US/docs/Bar /en-US/docs/Foo
```
I think it notice the infinite loop and it should now throw an error. 
I can file follow-ups for these things.

### CODE REVIEW

**Comment by peterbe** [comment]
Note: Once this lands and gets into `@mdn/yari` I'll update my hacky attempt here: https://github.com/mdn/content/pull/25 
and then I can proceed to write a GitHub Action workflow that actually fails. 

@escattone There's definitely an impending todo here about writing about this in the contribution guide. Yay!

**Comment by peterbe** [comment]
@fiji-flo I think you should take a look at this PR. It's currently unrelated to the fundamental redirects but it's a nice little project on its own. And if we can land this and follow-up with the other things I would like to do with validating the `_redirects.txt`, this might come in handy as a way to invoke the fundamental redirects. 

Can you perhaps take a little peek at the code itself and see how it feels?

**Comment by peterbe** [comment]
Note-to-self; https://github.com/mdn/yari/issues/404 is about improving redirects that can be improved. 
I just don't think that https://github.com/mdn/yari/issues/404 will solve *this* issue immediately. It might be better to do in a follow-up PR that digs deeper into the redirects beyond basic string checking.

File: redirects/cli.js
Comment (peterbe) [inline redirects/cli.js:11]: Dead code

File: redirects/cli.js
Comment (peterbe) [inline redirects/cli.js:24]: Dead code

File: content/redirect.js
Comment (peterbe) [inline content/redirect.js:55]: Dead code

**Review by escattone — APPROVED** [review]
@peterbe Looks good and seems to work just fine (`yarn redirects validate` found an error in my outdated `_redirects.txt`).

**Comment by peterbe** [comment]
https://github.com/mdn/content/pull/25 is a good continuation of this now that this CLI tool is in `@mdn/yari`.

**Pull request merged**
